### ISSUE TITLE
RuleRequest: prefer_else_if

### ISSUE THREAD

**Issue by lo1tuma** [realm/SwiftLint#4667]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.
   
When an `else` branch has only one child statement which is either an if or if-else statement, this could be refactored to an `else-if` statement.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Violating Examples:

```swift
if foo {
  bar()
} else {
  if baz {
    qux()
  }
}

if foo {
  bar()
} else {
  if baz {
    qux()
  } else {
    quux()
  }
}
```

Non-Violating Examples:

```swift
if foo {
  bar()
} else {
  if baz {
    qux()
  }
  somethingElse()
}

if foo {
  bar()
} else if baz {
  qux()
}

if foo {
  bar()
} else if {
  qux()
} else {
  quux()
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

This rule doesn’t need to be configurable.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

I think this should be an opt-in rule, because it is more like a stylistic rule to recommends to avoid unnecessary nesting.
